### ISSUE TITLE
Allow scheduler to wake up periodically

### ISSUE THREAD

**Issue by sallyruthstruik** [agronholm/apscheduler#305]
Hello. I work on django-apscheduler interface, and I faced the problem: when apscheduler decides that there are no jobs to execute, it sleeps until add_job will be called. So I faced this bug: https://github.com/jarekwg/django-apscheduler/issues/19

I think it is not good behavior, and in some cases programmer may want apscheduler to wake up periodically and check for a jobs. 
I suggest to declare some constant which apscheduler will return by default in _process_jobs method if something going wrong. 

If you agree with such behavior, I can provide pull request.

**Comment by agronholm** [comment]
Wow, looks like this one slipped through. Anyway, APScheduler was designed so it has complete control over the process of adding jobs to job stores. Therefore periodically checking does not make any sense, as there cannot be any new jobs it doesn't know about. If you're thinking about adding jobs directly to job stores, the FAQ has an entry which says this is not supported.

**Comment by sbernardes-convoso** [comment]
I'm a consumer of the library and recently faced the following problem:

Our architecture uses a farm of [1] producers and consumers and [2] a farm of only producers. Meaning that [2] will add jobs, but will never lookup/fetch and execute the jobs, whereas [1] will add and consume jobs.
This leads to a problem where if [1] looks up the next job in 3 hours and [2] adds a job to be run in 1 hour, the job will never run since [1] will sleep for 3 hours.

I was able to solve this by monkey-patching the _process_jobs function of the scheduler like so:
```python
def return_default_value_of_60(f):
    max_sleep_time_in_seconds = 60

    @wraps(f)
    def default_value_if_none(*a, **kw):
        return_value = f(*a, **kw)
        if return_value is None:
            return max_sleep_time_in_seconds
        return min(return_value, max_sleep_time_in_seconds)

    return default_value_if_none


BackgroundScheduler._process_jobs = return_default_value_of_60(BackgroundScheduler._process_jobs)
```

Shouldn't we reconsider having a maximum sleep time that does something similar as what I did?
Thank you in advance.
